### PR TITLE
feat: Makefile for consistent python testing interface

### PR DESCRIPTION
**Pull request by muncus**
## Description

This is a work in progress.

Create a Makefile that can be used to provide a consistent interface to running
samples, regardless of language.

### CODE REVIEW

**Comment by m-strzelczyk** [comment]
Would it be possible to add some kind of `init` command that would:

* create a virtualenv
* install `nox` and `black` in this virtualenv
* Use this virtualenv for other commands?

Right now, if I tried to use this makefile on a fresh git clone, I might get errors about missing nox binary etc.

At least, there should be some instructions that for this to work, there has to be `nox` in `$PATH` and `black` installed for the `nox -s blacken` command

**Comment by muncus** [comment]
We can certainly ensure that nox and black are pip-installed before other actions are called.

I'm wary of creating and using a virtualenv in the Makefile itself, because we would need to be  aware of the user's OS and shell - Would it be sufficient to surface a warning if run outside of a virtualenv?

**Comment by m-strzelczyk** [comment]
Yeah, a warning, with a link to some information on how to set up a virtualenv would be great.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+lint: noxfile.py
+	pip install nox black
+	cd ${dir}
+	nox $(NOXFLAGS) -s blacken
+	nox $(NOXFLAGS) -s lint
+
+# if no noxfile is present, we create one from the toplevel noxfile-template.py
+noxfile.py:
+	cd ${dir}
+	cp ${repo_root}/noxfile-template.py noxfile.py
```
Comment (m-strzelczyk) [inline Makefile:41]: I think this will just overwrite the existing noxfile.py, no matter if it's there or not. I think you want to add `-n` flag here.

**Review by m-strzelczyk — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (muncus) [reply Makefile]: I *think* Make will do the right thing when a noxfile.py exists already, but there's no harm in the -n, so i've added it anyway just to be safe

**Comment by muncus** [comment]
Without a python virtualenv, the user will now receive the following warning:

```
$ make lint dir=translate/samples/snippets/
Makefile:48: Use of a Python Virtual Environment is recommended. See README.md for details.
```

**Pull request merged**